server/routes/api/representatives.js

```diff
@@ -89,114 +90,169 @@ router.get('/:searchText', async (req, res) => {
+    {
       params,
       paramsSerializer: (params) =>
         qs.stringify(params, { arrayFormat: 'repeat' }),
-
       cache: {
         // TODO: we are disabling the cache to test results first
         ttl: 1000 * 60 * 60 * 24 * 7 // the time until the cached value is expired in milliseconds: set to 1 week
       }
-    })
-
-    // if you want to check if response was cached, uncomment: this is a way to track the issue
-    // console.log('isCached:', cached)
-
-    const { errors, results } = response
-    if (errors.length > 0) {
-      throw new Error(errors.join(','))
-    }
-    if (
-      !results ||
-      !Array.isArray(results.candidates) ||
-      results.candidates.length === 0
-    ) {
-      throw new Error('No matches found for the search criteria')
-    }
+    } // config
+  )
 
-    const officials = results.candidates[0].officials || []
+  if (filter != null) {
+    params.district_type = JURISDICTION_FILTER_MAP[filter]
+  }
 
-    const representatives = officials
-      .filter((rep) => {
-        // skip President and VP
-        return !(
-          rep.office.district.ocd_id === 'ocd-division/country:us' &&
-          /^(Vice )?President$/.test(rep.office.title)
-        )
-      })
-      .map((rep) => {
-        const mainAddr =
-          (Array.isArray(rep.addresses) && rep.addresses[0]) || {}
-
-        const repInfo = {
-          id: rep.sk || rep.id || '',
-          name: formatName(rep),
-          title:
-            rep.office.title ||
-            rep.office.chamber.name_formal ||
-            rep.office.chamber.name ||
-            '',
-          address_line1: mainAddr.address_1 || '',
-          address_line2:
-            mainAddr.address_2 +
-              (mainAddr.address_3 ? ', ' + mainAddr.address_3 : '') || '',
-          address_city: mainAddr.city || '',
-          address_state: mainAddr.state || '',
-          address_zip: mainAddr.postal_code || '',
-          address_country: 'US',
-          email:
-            (Array.isArray(rep.email_addresses) && rep.email_addresses[0]) ||
-            'Not Made Public',
-          contactPage:
-            rep.web_form_url || (Array.isArray(rep.urls) && rep.urls[0]) || '',
-          photoUrl:
-            rep.photo_origin_url ||
-            'https://cdn.pixabay.com/photo/2016/08/08/09/17/avatar-1577909_1280.png',
-          socialMediaPages: getOfficialSocialMediaPages(rep.identifiers),
-          photoCroppingCSS: getPhotoCroppingValues(rep.photo_cropping)
+  let response
+
+  if (error) {
+    // http://127.0.0.1:8080/api/representatives/91765 get request failed
+    // uncomment only when debugging
+    // console.error(error)
+
+    // to avoid server crashing so front-end work can continue, add NODE_ENV='development' to your .env
+    if (process.env.NODE_ENV === 'development') {
+      // a test_response helpful for front-end work when axios requests are not working
+      const test_response = {
```

Comment:
This is the variable that allows frontend to move to the next step even when getting the representatives request failed. I figured we only want to do this when we are in development mode.

Reply:
@yoyoyojoe Good point! Do you think this should be related to #509? Maybe for `NODE_ENV === 'development'` we can choose to hit the mock endpoint instead of the actual API.